### PR TITLE
Enhancements for crypto

### PR DESCRIPTION
**Pull request by A-A-G**
https://forum.etoa.ch/index.php?page=Thread&postID=112859#post112859

### CODE REVIEW

File: htdocs/content/crypto.php [hunk 1]
```diff
@@ -193,75 +193,85 @@
...
+												FROM
+													fleet
+												WHERE
+													entity_from=".$target->id()."
+													OR entity_to=".$target->id()."");
+												$out.="Es sind ".mysql_num_rows($fres)." Flotten unterwegs.\n\n";
+											}
+											else
+											{
+												$out.="Es sind Flotten unterwegs.\n\n";
```
Comment (glaubinix) [inline htdocs/content/crypto.php:216]: This now shows `Es sind Flotten unterwegs.` if the `$decryptlevel` even if there are no fleets which is wrong information

File: htdocs/content/crypto.php [hunk 2]
```diff
@@ -193,75 +193,85 @@
...
-															$rand = random_int(0, 2*7*60);
-															$out.="Zwischen ".date("d.m.Y H:i",$fd->landTime()-$rand)." und ".date("d.m.Y H:i",$fd->landTime()+(2*7*60)-$rand)." Uhr";
+															$out.= "\n[b]Ankunft:[/b] ";
+															$out.=date("d.m.Y H:i",$fd->landTime())." Uhr";
 														}
-														elseif ($decryptlevel<=25)
+														elseif ($decryptlevel>=$cfg->value("crypto_time_15_level"))
 														{
-															$out.=date("d.m.Y H:i",$fd->landTime())." Uhr";
+															$rand = random_int(0, 2*7*60);
```
Comment (glaubinix) [inline htdocs/content/crypto.php:252]: This is a 14min timeframe but the config says 15

File: htdocs/content/crypto.php [hunk 3]
```diff
@@ -193,75 +193,85 @@
...
+															$out.= "\n[b]Ankunft:[/b] ";
+															$out.="Zwischen ".date("d.m.Y H:i",$fd->landTime()-$rand)." und ".date("d.m.Y H:i",$fd->landTime()+(2*7*60)-$rand)." Uhr";
 														}
-														else
+														elseif ($decryptlevel>=$cfg->value("crypto_time_30_level"))
 														{
-															$out.=date("d.m.Y H:i:s",$fd->landTime())." Uhr";
+															$rand = random_int(0, 30*60*2);
```
Comment (glaubinix) [inline htdocs/content/crypto.php:258]: This is a 60min timeframe but the config says 30

File: htdocs/content/crypto.php [hunk 4]
```diff
@@ -270,125 +280,161 @@
...
 														{
-															$out.="Zwischen ".date("d.m.Y H:i",$fd->landTime()-(7*60))." und ".date("d.m.Y H:i",$fd->landTime()+(7*60))." Uhr";
+															$out.= "\n[b]Ankunft:[/b] ";
+															$out.=date("d.m.Y H:i",$fd->landTime())." Uhr";
 														}
-														elseif ($decryptlevel<=25)
+														elseif ($decryptlevel>=$cfg->value("crypto_time_15_level"))
 														{
-															$out.=date("d.m.Y H:i",$fd->landTime())." Uhr";
+															$rand = random_int(0, 2*7*60);
```
Comment (glaubinix) [inline htdocs/content/crypto.php:367]: Same 14min timeframe but config says 15

File: htdocs/content/crypto.php [hunk 5]
```diff
@@ -270,125 +280,161 @@
...
+															$out.= "\n[b]Ankunft:[/b] ";
+															$out.="Zwischen ".date("d.m.Y H:i",$fd->landTime()-$rand)." und ".date("d.m.Y H:i",$fd->landTime()+(2*7*60)-$rand)." Uhr";
 														}
-														else
+														elseif ($decryptlevel>=$cfg->value("crypto_time_30_level"))
 														{
-															$out.=date("d.m.Y H:i:s",$fd->landTime())." Uhr";
+															$rand = random_int(0, 30*60*2);
```
Comment (glaubinix) [inline htdocs/content/crypto.php:373]: Same 60min timeframe but config says 30

File: htdocs/content/crypto.php [hunk 6]
```diff
@@ -193,75 +193,85 @@
 										// Do the scan if chance >= 0
 										if ($chance >= 0)
```
Comment (glaubinix) [inline htdocs/content/crypto.php:194]: Do we need to adjust the calculation of `$chance` as well? It has the same `mt_rand(0,2)-1` calculation in there

File: htdocs/content/crypto.php (hunk 2 above)
Comment (A-A-G) [reply htdocs/content/crypto.php:252]: 15 is taken from the old code comments and the random modificator is unchanged. I don't know why there wass written +- 15 mins for 14. Just changing 2*7 to 15 would keep it an int.

File: htdocs/content/crypto.php (hunk 6 above)
Comment (A-A-G) [reply htdocs/content/crypto.php:194]: I'm not sure if that's necessary. At least for 3.6. For me, success chance was always ok and better balanced than the overall results after passing this threshold.

**Review by glaubinix — APPROVED** [review]

**Pull request merged**